File: 구름 알고리즘 미션/동적 프로그래밍/피보나치 수열/피보나치 수열.js
const MOD = (10**9) + 7;
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const N = +input[0];
const dp = Array(N+1).fill(0);

dp[1] = 0;
dp[2] = 1;

for(let i = 3 ; i <= N ; i++){
	dp[i] = (dp[i-1] + dp[i-2]) % MOD;
}

console.log(dp[N]);

File: 구름 알고리즘 미션/동적 프로그래밍/피보나치 수열/피보나치 수열.py
import sys
input = sys.stdin.readline
MOD = (10**9) + 7

N = int(input())
dp = [0] * (N + 1)
dp[1], dp[2] = 0, 1

for i in range(3, N+1):
	dp[i] = (dp[i-1] + dp[i-2]) % MOD

print(dp[N])
